### PR TITLE
Show an informational warning before performing hierarchy operations

### PR DESCRIPTION
**Pull request by kennsippell**
#232 

image

### CODE REVIEW

File: src/lib/manage-hierarchy.ts [hunk 1]
```diff
@@ -41,26 +38,57 @@ export default class ManageHierarchyLib {
...
-
-  const { sourceId, destinationId } = getSourceAndDestinationIds(hierarchyAction, sourceLineage, destinationLineage);
-  const jobData = getJobData(hierarchyAction, sourceId, destinationId, chtApi);
-  const jobName = getJobName(jobData.action, sourceLineage, destinationLineage);
-  const jobParam: JobParams = {
-    jobName,
-    jobData,
-  };
+function diffNowInDays(dateTime: DateTime): number {
+  return -(dateTime?.diffNow('days')?.days || 0);
```
[the remark this answers is not in the document]
Comment (kennsippell) [reply src/lib/manage-hierarchy.ts:74]: Actually the Luxon library's DateTime.diffNow function returns negatives and that negative makes them positive

File: src/lib/cht-api.ts [hunk 2]
```diff
@@ -134,6 +135,20 @@ export class ChtApi {
     return this.axiosInstance.post(url, deactivationPayload);
   };
 
+  countContactsUnderPlace = async (docId: string): Promise<number> => {
+    const url = `medic/_design/medic/_view/contacts_by_depth`;
+    console.log('axios.get', url);
```
Comment (binokaryg) [inline src/lib/cht-api.ts:140]: Why are we logging a constant? Do we wish to keep it for a while?

Comment (kennsippell) [reply src/lib/cht-api.ts:140]: I personally like seeing when the server makes requests to CHT, so all outbound requests from this cht-api library are logged. Fred has also indicated that he didn't like this pattern, but for now I'll stay consistent with the file and we can remove everywhere if we wish to remove.

File: src/lib/manage-hierarchy.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  }
+
+  if (hierarchyAction === 'delete') {
+    return { sourceId, destinationId: '' };
+  }
+
+  const destinationIndex = hierarchyAction === 'move' ? 1 : 0;
+  const destinationId = destinationLineage[destinationIndex]?.id;
+  if (!destinationId) {
+    throw Error('Unexpected error: Hierarchy operation due to missing destination information');
```
Comment (binokaryg) [inline src/lib/manage-hierarchy.ts:111]:
```suggestion
    throw Error('Unexpected error: Hierarchy operation failed due to missing destination information');
```

File: src/services/hierarchy-view-model.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import _ from 'lodash';
+import { Config, ContactType } from '../config';
+import ManageHierarchyLib from '../lib/manage-hierarchy';
+
+export function hierarchyViewModel(action: string, contactType: ContactType) {
+  const parentTypeName = contactType.hierarchy.find(h => h.level === 1)?.contact_type;
+  if (!parentTypeName) {
+    throw Error('parent type name');
```
Comment (binokaryg) [inline src/services/hierarchy-view-model.ts:8]:
```suggestion
    throw Error('Parent type name not found in contact type hierarchy');
```

**Review by binokaryg — APPROVED** [review]

**Review by binokaryg — APPROVED** [review]
Approving with a few optional suggestions.

**Pull request merged**